x-pack/plugin/sql/src/main/java/org/elasticsearch/xpack/sql/optimizer/Optimizer.java

```diff
@@ -484,6 +509,247 @@ protected LogicalPlan rule(OrderBy ob) {
         }
     }
 
+    static class CombineFilters extends OptimizerRule<Filter> {
+
+        @Override
+        protected LogicalPlan rule(Filter plan) {
+            // merge the current filter with a Filter above a Limit/Pivot/Aggregate
+            LogicalPlan maybeFilterChild = plan.child().collectOnlyChild(
+                c -> c instanceof Filter || c instanceof Limit || c instanceof Pivot || c instanceof Aggregate);
+
+            LogicalPlan result = plan;
+            if (maybeFilterChild instanceof Filter) {
+                Filter childFilter = (Filter) maybeFilterChild;
+                // merge the filter with the child filter, move the condition down into the childFilter
+                result = plan.child().transformDown(p -> {
+                    if (p == childFilter) {
+                        return new Filter(Source.EMPTY, childFilter.child(),
+                            new And(Source.EMPTY, plan.condition(), childFilter.condition()));
+                    } else {
+                        return p;
+                    }
+                });
+            }
+            return result;
+        }
+    }
+    
+    public static class PushDownFilters extends OptimizerRule<Filter> {
+
+        PushDownFilters() {
+            super(TransformDirection.UP);
+        }
+
+        @Override
+        protected LogicalPlan rule(Filter plan) {
+            // Filter[condition=f(grouping, aggregate)] -> ... -> Aggregate structure will be broken up into
+            // two separate filters if possible, so the filter on the grouping can be pushed down
+            // SELECT groupings, aggregates, ... FROM ... GROUP BY [groupings]
+            // the aggregates in this context mean only the aggregates that are not part of the grouping
```

Comment:
Quoting might improve a bit the legibility.
```suggestion
            // the "aggregates" in this context mean only the aggregates that are not part of the grouping
```